ofctrl/ofAction.go

```diff
@@ -778,12 +781,38 @@ type SetICMPv4CodeAction struct {
 	Code uint8
 }
 
-func (a *SetICMPv4CodeAction) GetActionMessage() openflow13.Action {
-	field, _ := openflow13.FindFieldHeaderByName("NXM_OF_ICMP_CODE", false)
-	field.Value = &openflow13.IcmpCodeField{Code: a.Code}
-	return openflow13.NewActionSetField(*field)
+func (a *SetICMPv4CodeAction) GetActionMessage() openflow15.Action {
+	field, _ := openflow15.FindFieldHeaderByName("NXM_OF_ICMP_CODE", false)
+	field.Value = &openflow15.IcmpCodeField{Code: a.Code}
+	return openflow15.NewActionSetField(*field)
 }
 
 func (a *SetICMPv4CodeAction) GetActionType() string {
 	return ActTypeSetICMP4Code
 }
+
+type CopyFieldAction struct {
+	SrcOxmId openflow15.OxmId
+	DstOxmId openflow15.OxmId
+	NBits uint16
+	SrcOffset uint16
+	DstOffset uint16
+}
+
+func (a *CopyFieldAction) GetActionMessage() openflow15.Action {
+	return openflow15.NewActionCopyField(a.NBits, a.SrcOffset, a.DstOffset, a.SrcOxmId, a.DstOxmId)
+}
+
+func (a *CopyFieldAction) GetActionType() string {
+	return ActTypeCopyField
+}
+
+func NewCopyFieldAction(nBits uint16, srcOffset uint16, dstOffset uint16, srcOxmId *openflow15.OxmId, dstOxmId *openflow15.OxmId) (*CopyFieldAction, error) {
```

Comment:
`ResetFieldsLength` is not supported in this new OFAction `CopyFieldAction`, it is possible to introduce issue when we expected to copy value from register to tun_metadataX. Function `ResetFieldsLength` resets tun_metadataX field length the with the length in TLV map settings.

Reply:
I will add the implementation of ResetFieldsLength in CopyFieldAction.